internal/publisher/core.go

```diff
@@ -43,6 +44,15 @@ func (p *Core) Publish(ctx context.Context, req *metrov1.PublishRequest) ([]stri
 
 	msgIDs := make([]string, 0)
 
+	orderingKey := ""
+	if len(req.Messages) > 0 {
+		orderingKey = req.Messages[0].OrderingKey
```

Comment:
// All messages in request will have the same ordering key 

^ Didn't get this restriction, why we want to put this restriction ?